internal/blocksync/reactor.go

```diff
@@ -109,6 +111,7 @@ func NewReactor(state sm.State, blockExec *sm.BlockExecutor, store *store.BlockS
 		store:        store,
 		pool:         pool,
 		blockSync:    blockSync,
+		myAddr:       myAddr,
```

Comment:
I can't believe that we don't know our validator address in this protocol...

Reply:
We never needed it so far

Reply:
When I am not a validator, what do we get here?

Reply:
All nodes have a `privValidator`: notice that this [line](https://github.com/cometbft/cometbft/blob/9742bf6e1/node/node.go#L349) is run by all full nodes. The only difference is that if you are not a validator, that address doesn't appear in the validator set